crates/starknet/src/starknet/mod.rs

```diff
@@ -580,19 +621,24 @@ mod tests {
     #[test]
     fn getting_state_reader_at_block_by_hash() {
         let config = starknet_config_for_test();
-        let starknet = Starknet::new(&config).unwrap();
-        match starknet.get_state_at(&BlockId::Number(2)) {
-            Err(Error::BlockIdNumberUnimplementedError) => (),
+        let mut starknet = Starknet::new(&config).unwrap();
+        starknet.generate_new_block(StateDiff::default(), starknet.state.clone()).unwrap();
+
+        match starknet.get_state_at(&BlockId::Hash(Felt::from(0).into())) {
+            Err(Error::NoBlock) => (),
             _ => panic!("Should have failed"),
         }
     }
 
     #[test]
     fn getting_state_reader_at_block_by_number() {
```

Comment:
Same as for `getting_state_reader_at_block_by_hash`

Reply:
[8757a47](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/104/commits/8757a47ecd6e798fc5c85f4251a68aa8734087c0)